tests/api.c

```diff
@@ -18304,6 +18304,72 @@ static int test_wc_PKCS7_SetAESKeyWrapUnwrapCb(void)
     return EXPECT_RESULT();
 }
 
+/*
+ * Testing wc_PKCS7_GetEnvelopedDataKariRid().
+ */
+static int test_wc_PKCS7_GetEnvelopedDataKariRid(void)
+{
+    EXPECT_DECLS;
+#if defined(HAVE_PKCS7)
+#if defined(HAVE_ECC) && (!defined(NO_AES) || (!defined(NO_SHA) || \
+     !defined(NO_SHA256) || defined(WOLFSSL_SHA512)))
+    byte rid[256];
+    byte cms[1024];
+    XFILE cmsFile = XBADFILE;
+    int ret;
+    word32 ridSz = sizeof(rid);
+    XFILE skiHexFile = XBADFILE;
+    byte skiHex[256];
+    word32 cmsSz;
+    word32 skiHexSz;
+    size_t i;
+    const word32 ridKeyIdentifierOffset = 4;
+
+    cmsFile = XFOPEN("./certs/test/kari-keyid-cms.msg", "rb");
+    ExpectTrue(cmsFile != XBADFILE);
+    cmsSz = (word32)XFREAD(cms, 1, sizeof(cms), cmsFile);
+    ExpectTrue(cmsSz > 0);
+    if (cmsFile != XBADFILE)
+        XFCLOSE(cmsFile);
+
+    skiHexFile = XFOPEN("./certs/test/client-ecc-cert-ski.hex", "rb");
+    ExpectTrue(skiHexFile != XBADFILE);
+    skiHexSz = (word32)XFREAD(skiHex, 1, sizeof(skiHex), skiHexFile);
+    ExpectTrue(skiHexSz > 0);
+    if (skiHexFile != XBADFILE)
+        XFCLOSE(skiHexFile);
+
+    ret = wc_PKCS7_GetEnvelopedDataKariRid(cms, cmsSz, rid, &ridSz);
+    ExpectIntEQ(ret, 0);
+    ExpectIntGT(ridSz, ridKeyIdentifierOffset);
+    /* The Subject Key Identifier hex file should have 2 hex characters for each
+     * byte of the key identifier in the returned recipient ID (rid), plus a
+     * terminating new line character. */
+    ExpectIntGE(skiHexSz, ((ridSz - ridKeyIdentifierOffset) * 2) + 1);
```

Comment:
Careful here. This is an unsigned, so computation may not be valid if ridSz is less than equal to 5.